### PR TITLE
Property handle zero duration animation clips(2nd fix)

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #11879

### Changelog

* In #11879 I proposed a PR to handle zero duration animation, but it doesn't seems correct since single frame clips(created by editor, having zero duration) can not be sampled. In this PR I'll try to strike this.

After this PR, given zero duration clip `C`, an animation state `S` of `C`, the following should always hold:

- As long as the `S`'s playing or `S.sample()` is called, the clip `C` will be sampled at time 0.

  > In #11879 this is not true!

- In anytime, `S.current === S.ratio === 0`.

- In anytime, if "wrap info" is inspected on `S`(from within other engine code), the wrap info's following properties keeps constant:

  - `ratio` - 0.0
  - `time` - 0.0
  - `iteration` - 1.0(which means the starting point of first iteration)
  - `direction` - 1.0(which means forwarding iteration)
  - `stopped` - `true` if `S.repeatCount` is finite, `false` otherwise. In other words, any single update invocation on `S` stops `S` no matter how long the update delta time, if `S` has finite repeat count(for example, non-loop wrap mode, manually set finite repeat count). In other case, `S` keeps loop forever.
    >  In #11879 `stopped` would be always `true`.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by shrinktofit** [comment]
@cocos-robot run test cases

File: cocos/core/animation/animation-state.ts [hunk 1]
```diff
@@ -494,9 +494,6 @@ export class AnimationState extends Playable {
 
     public sample () {
         const info = this.getWrappedInfo(this.time, this._wrappedInfo);
-        if (this._playbackDuration === 0.0) {
```
Comment (shrinktofit) [inline cocos/core/animation/animation-state.ts:497]: Added in #11879

File: cocos/core/animation/animation-state.ts [hunk 2]
```diff
@@ -582,14 +579,17 @@ export class AnimationState extends Playable {
     private simpleProcess () {
         const playbackStart = this._playbackRange.min;
         const playbackDuration = this._playbackDuration;
-        if (playbackDuration === 0.0) {
```
Comment (shrinktofit) [inline cocos/core/animation/animation-state.ts:585]: Added in #11879

File: tests/animation/animation-state.test.ts [hunk 3]
```diff
@@ -28,18 +28,22 @@ describe('Animation state', () => {
         // - `time`  - 0.0
         // - `iteration` - 1.0(starting point of first iteration)
         // - `direction` - 1.0(forwarding iteration)
+        // - `stopped` - true if `.repeatCount` is finite false otherwise
         // Zero duration states won't do sample even there are curves in zero time.
         expectState(state);
         expectWrappedInfo(state.sample());
         blendStateBuffer.apply();
-        expect(node.position.x).toBeCloseTo(0.0);
+        expect(node.position.x).toBeCloseTo(2.0);
```
Comment (shrinktofit) [inline tests/animation/animation-state.test.ts:36]: It should be 2.0 as this PR mainly to solve with

**Review by SantyWang — APPROVED** [review]

**Pull request merged**